editoast/src/views/rolling_stocks/mod.rs

```diff
@@ -581,429 +572,492 @@ async fn create_compound_image(
 
 #[cfg(test)]
 pub mod tests {
+    use std::ops::DerefMut;
     use std::vec;
 
     use actix_http::Request;
     use actix_http::StatusCode;
-    use actix_web::dev::ServiceResponse;
     use actix_web::http::header::ContentType;
+    use actix_web::test::call_and_read_body_json;
     use actix_web::test::call_service;
     use actix_web::test::read_body_json;
     use actix_web::test::TestRequest;
+    use pretty_assertions::assert_eq;
     use rstest::rstest;
     use serde_json::json;
-    use std::sync::Arc;
 
-    use super::retrieve_existing_rolling_stock;
     use super::RollingStockError;
     use super::TrainScheduleScenarioStudyProject;
     use crate::assert_response_error_type_match;
-    use crate::assert_status_and_read;
-    use crate::fixtures::tests::db_pool;
-    use crate::fixtures::tests::get_fast_rolling_stock_form;
-    use crate::fixtures::tests::get_other_rolling_stock_form;
-    use crate::fixtures::tests::named_fast_rolling_stock;
-    use crate::fixtures::tests::named_other_rolling_stock;
+    use crate::error::InternalError;
     use crate::fixtures::tests::train_schedule_with_scenario;
+    use crate::modelsv2::fixtures::create_fast_rolling_stock;
+    use crate::modelsv2::fixtures::create_rolling_stock_with_energy_sources;
+    use crate::modelsv2::fixtures::fast_rolling_stock_changeset;
+    use crate::modelsv2::fixtures::fast_rolling_stock_form;
+    use crate::modelsv2::fixtures::get_rolling_stock_with_invalid_effort_curves;
+    use crate::modelsv2::fixtures::rolling_stock_with_energy_sources_form;
     use crate::modelsv2::prelude::*;
-    use crate::modelsv2::rolling_stock_model::tests::get_invalid_effort_curves;
     use crate::modelsv2::rolling_stock_model::RollingStockModel;
-    use crate::modelsv2::DbConnectionPool;
     use crate::views::rolling_stocks::rolling_stock_form::RollingStockForm;
-    use crate::views::rolling_stocks::RollingStockKey;
+    use crate::views::test_app::TestAppBuilder;
     use crate::views::tests::create_test_service;
 
+    pub fn rolling_stock_create_request(rolling_stock_form: &RollingStockForm) -> Request {
+        TestRequest::post()
+            .uri("/rolling_stock")
+            .set_json(rolling_stock_form)
+            .to_request()
+    }
+
+    pub fn rolling_stock_get_by_id_request(rolling_stock_id: i64) -> Request {
+        TestRequest::get()
+            .uri(format!("/rolling_stock/{rolling_stock_id}").as_str())
+            .to_request()
+    }
+
+    pub fn rolling_stock_delete_request(rolling_stock_id: i64) -> Request {
+        TestRequest::delete()
+            .uri(format!("/rolling_stock/{rolling_stock_id}").as_str())
+            .to_request()
+    }
```

Comment:
These functions probably don't need to be `pub`? (I didn't check if they're used somewhere else but I would find it a weird pattern for a `mod tests` to expose an API).